### ISSUE TITLE
Strategy calibration analysis: all strategies losing except mean_reversion fluke

### ISSUE THREAD

**Issue by deanturpin** [deanturpin/lft#10]
# Strategy Calibration Analysis: Only Mean Reversion Enabled

## Current Situation

After 3-day calibration on 1-minute bars, only **mean_reversion** is enabled with +$2.44 P&L from a single trade.

```
✓ ma_crossover      Complete: 3657 signals, 2376 trades, $-708.67 P&L, 35.6% WR
✓ volatility_breakout Complete: 1783 signals, 113 trades, $-94.10 P&L, 27.4% WR  
✓ mean_reversion    Complete: 1 signals, 1 trades, $2.44 P&L, 100.0% WR
✓ relative_strength Complete: 263 signals, 202 trades, $-144.22 P&L, 33.7% WR
✓ volume_surge      Complete: 130 signals, 126 trades, $-83.45 P&L, 33.3% WR
```

**Result**: System runs live with essentially no trading (1 signal in 3 days = ~1-2 trades/week).

---

### Problem 1: Mean Reversion's "Success" Is A Statistical Fluke

- Only **1 signal** in 40,000+ opportunities (35 symbols × 3 days × 390 minutes)
- Strategy requires price >2 standard deviations below MA (extremely rare condition)
- **100% win rate with n=1 is meaningless** - pure luck this trade worked
- Not a viable live trading strategy at this signal frequency

### Problem 2: All Other Strategies Losing Money

**Why they're failing:**

#### 1. Stop Losses Too Tight For 1-Minute Timeframe
- **Current**: 1.0% stop loss
- **Reality**: Average intrabar noise is 0.3-0.8%
- **Result**: Getting stopped out on normal volatility (whipsaw)

**Example scenario:**
```
10:00: Buy AAPL at $150.00 (signal triggers)
10:01: Price dips to $148.60 (-0.93%) on normal noise
10:02: Stop loss hit at $148.50 (-1.0%) → -$10 loss
10:05: Price recovers to $151.20 (+0.8%) → would have been profitable
```

This is textbook **whipsaw** - tight stops killing trades that would have won.

#### 2. Strategies Designed For Daily/Hourly Bars, Not Minutes
- **MA crossover**: Works on H1/D1 timeframes, generates noise on M1
- **Mean reversion**: Needs hours/days to revert, not minutes
- **Volatility breakout**: Needs sustained moves (hours), not 1-minute spikes
- **Relative strength**: Momentum plays out over hours/days
- **Volume surge**: Institutional flow takes time to complete

**We're running daily strategies on minute data** → high noise, low edge.

#### 3. No Market Regime Filtering
All strategies fire in all market conditions:
- **Choppy/ranging markets** → whipsaw central (MA crossover killed)
- **Low volume** → wide spreads eat edge
- **High volatility** → stops too tight (volatility breakout fails)
- **Trending markets** → actually works! (but rare)

---

### MA Crossover: -$708.67 (worst performer)
- **3,657 signals** → 2,376 trades (65% conversion - high execution rate)
- **35.6% win rate** (barely better than random)
- **Problem**: In ranging markets (most of the time):
  - Short MA crosses above long MA → BUY
  - Price reverses immediately → Stop loss hit
  - Repeat endlessly → death by 1000 cuts

### Volatility Breakout: -$94.10
- **1,783 signals** → 113 trades (**6.3% conversion** - very low!)
- **27.4% win rate**
- **Problem**: Requires positive momentum + volatility expansion
  - Many signals blocked by existing positions
  - Breakouts often false (reverse immediately)
  - 1% stop too tight for volatile moves

### Relative Strength: -$144.22
- **263 signals** → 202 trades (76.8% conversion)
- **33.7% win rate**
- **Problem**: "Buy the winners" in high-frequency is dangerous
  - By the time we detect outperformance, often already peaked
  - Momentum exhaustion
  - 1% stop doesn't give winners room to continue

### Volume Surge: -$83.45
- **130 signals** → 126 trades (**96.9% conversion** - excellent detection)
- **33.3% win rate**
- **Problem**: Volume spikes often indicate:
  - News events → big moves that reverse quickly
  - Institutional dumping → buying the top
  - Pump and dumps → artificial volume
- High conversion but low win rate = good signal detection, bad timing/exits

---

### Option 1: Widen Stop Losses (Quick Fix) ⚡
**Change**: Increase from 1% to 2.0-2.5% for 1-minute bars

```cpp
// include/exit_logic_tests.h
constexpr auto stop_loss_pct = 2.5_pc;  // Give trades room to breathe
```

**Expected impact**: 
- Win rates should improve to 45-50%
- 2-3 strategies should become profitable
- Re-run calibration to verify

**Effort**: 5 minutes
**Risk**: Low - just testing wider stops

---

### Option 2: Use Longer Timeframes (Better Fix) 📊
**Change**: Switch calibration from 1-minute to 5-minute or 15-minute bars

```cpp
// lft.cxx:393
auto bars = is_crypto 
    ? client.get_crypto_bars(symbol, "5Min", start, end)  // Was "1Min"
    : client.get_bars(symbol, "5Min", start, end);
```

**Expected impact**:
- Less noise → strategies work as designed
- Better signal quality
- Fewer whipsaws
- Need to adjust live mode polling to match (5min bars = poll every 5min)

**Effort**: 30 minutes (change calibration + live polling frequency)
**Risk**: Medium - changes system behaviour significantly

---

### Option 3: Add Market Regime Filters (Best Fix) 🎯
**Change**: Only trade strategies in favorable conditions

```cpp
// strategies.cxx - add to each strategy
// Don't trade MA crossover in low-volatility ranges
if (strategy == "ma_crossover" and history.recent_noise(20) < 0.003)
    return signal;  // Too choppy

// Don't trade relative strength in downtrends
if (strategy == "relative_strength" and market_average < -0.2)
    return signal;  // Market selling off

// Don't trade breakouts in low-volume conditions  
if (strategy == "volatility_breakout" and history.volume_factor() > 1.2)
    return signal;  // Volume too low
```

**Expected impact**:
- Reduce bad signals in unfavorable conditions
- Improve win rates without changing core strategy logic
- More sophisticated approach

**Effort**: 2-3 hours (design filters, implement, test)
**Risk**: Medium - need to get regime detection right

---

### Option 4: ATR-Based Dynamic Stops (Advanced) 🔬
**Change**: Replace fixed % stops with ATR (Average True Range) based stops

```cpp
// Adapt stop loss to recent volatility
auto atr = history.average_true_range(14);  // 14-period ATR
auto stop_loss = entry_price - (2.0 * atr);  // 2x ATR stop

// Instead of fixed:
auto stop_loss = entry_price * (1.0 - stop_loss_pct);
```

**Benefits**:
- Tight stops in calm markets
- Wide stops in volatile markets
- Adapts to current conditions automatically

**Effort**: 4-6 hours (implement ATR calculation, modify exit logic, test)
**Risk**: High - changes fundamental exit mechanism

---

### Option 5: Accept Reality & Run Monitor-Only Mode
**Change**: Disable all strategies, just monitor positions

**Current state**: With only mean_reversion enabled (1 signal/3 days), we're essentially doing this anyway.

**Recommendation**: If going this route, explicitly disable trading and focus on:
- Fixing the stale trades bug (see #10 - if created)
- Improving display/monitoring
- Planning strategy overhaul

---

## Recommended Action Plan

**Immediate (Tonight/Tomorrow Morning):**
1. ✅ **Try Option 1**: Widen stops to 2.5%
   - Quick test, low risk
   - Re-run calibration
   - If 2+ strategies become profitable → ship it

**If Option 1 Doesn't Work:**
2. 📊 **Implement Option 2**: Switch to 5-minute bars
   - More fundamental fix
   - Strategies will work better
   - Adjust polling frequency to match

**Longer Term:**
3. 🎯 **Add Option 3**: Market regime filters
   - Don't fight unfavorable conditions
   - Improve edge significantly

4. 🔬 **Consider Option 4**: ATR-based stops
   - Most sophisticated approach
   - Adapts to market conditions

---

## Key Metrics To Watch

After implementing fixes, calibration should show:

| Metric | Current (Bad) | Target (Good) |
|--------|--------------|---------------|
| Strategies enabled | 1/5 (20%) | 3-4/5 (60-80%) |
| Average win rate | 35% | 45-50% |
| Signals per day | <1 | 20-50 |
| Mean reversion signals | 1 in 3 days | Should stay rare (it's extreme condition) |
| MA crossover P&L | -$708 | Positive or disabled |

---

## The Bottom Line

**Current setup is fundamentally broken**:
- 1% stops on 1-minute bars = guaranteed whipsaw
- Running daily strategies on minute timeframe = high noise, no edge
- Mean reversion's $2.44 "success" is statistical noise (n=1)

**System is effectively not trading** (1 signal per 3 days).

**Must fix** to have viable live trading system.
